src/mobs/spawn_mob.js

```diff
@@ -0,0 +1,102 @@
+import { mobs } from './mobs.js'
+import { chunk_position } from '../chunk.js'
+const mcData = minecraftData(version)
+
+
+const color_by_type = {
+  mob:'white',
+  archiMob:'gold',
+  boss:'red',
+  npc:'green',
+  garde:'blue'
+}
+
+
+//generate an object containing mobs by chunk
+console.log("spawn_mob.js -> TODO: fix mob id assignement");
+
+let mobs_in_ckunk = []
+
+for(let i=0;i<world_mobs_datas.length;i++) {
+  world_mobs_datas[i].id = i + 1000 // add mob unique id for this code
+  if(mobs_in_ckunk[chunk_position(world_mobs_datas[i].position.x) + ":" + chunk_position(world_mobs_datas[i].position.z)] == undefined){
+    mobs_in_ckunk[chunk_position(world_mobs_datas[i].position.x) + ":" + chunk_position(world_mobs_datas[i].position.z)] = []
+  }
+  mobs_in_ckunk[chunk_position(world_mobs_datas[i].position.x) + ":" + chunk_position(world_mobs_datas[i].position.z)].push(world_mobs_datas[i])
+
+  // check files validity
+  if(mobs[world_mobs_datas[i].mob] == undefined){
+    throw "mob " + world_mobs_datas[i].mob + " (file 'world_mobs_datas.json') is not defined in file  'mobs.js'"
+  }
+  else if(mcData.entitiesByName[mobs[world_mobs_datas[i].mob].mob] == undefined)
+  {
+    throw "unvalid 'mob' in " + JSON.stringify(mobs[world_mobs_datas[i].mob])
+  }
+  else if(color_by_type[mobs[world_mobs_datas[i].mob].type] == undefined)
+  {
+    throw "unvalid 'type' in " + JSON.stringify(mobs[world_mobs_datas[i].mob])
+  }
+}
+
+
+export function spawn_mob({ client, events, position }, server) {
+  events.on('chunk_loaded', ({x, z}) => {
+    if(mobs_in_ckunk[x + ":" + z] != undefined)
+    {
+      mobs_in_ckunk[x + ":" + z].map((m) => {
+        const mob = {
+          entityId: m.id,
+          entityUUID: UUID.v4(),
+          type: mcData.entitiesByName[mobs[m.mob].mob].id,
+          x: m.position.x,
+          y: m.position.y,
+          z: m.position.z,
+          yaw: 0,
+          pitch: 0,
+          headPitch: 0,
+          velocityX: 400,
+          velocityY: 400,
+          velocityZ: 400
+        }
+
+        const metadata = {
+          entityId: m.id,
+          metadata: [
+            {
+              key: 2,
+              value: JSON.stringify({text: mobs[m.mob].displayName, color: color_by_type[mobs[m.mob].type], extra: m.level && [{text: ` [ ${m.level} ] `, color: "dark_red"}]}),
+              type: 5
+            },
+            {
+              key: 3,
+              type: 7,
+              value: true
+            }
+          ]
+        }
+
+        client.write('spawn_entity_living', mob)
+        client.write('entity_metadata', metadata)
+      })
+    }
+  })
+
+  events.on('chunk_unloaded', ({x, z}) => {
+    if(mobs_in_ckunk[x + ":" + z] != undefined){
+      let entity_ids = []
+
+      mobs_in_ckunk[x + ":" + z].map((m) => {
+        entity_ids.push(m.id)
+      })
+
+      client.write('entity_destroy', {
+        entityIds: entity_ids
+      })
```

Comment:
When you map and array map will return you a new array so you can write this like


```suggestion
      client.write('entity_destroy', {
        entityIds: mobs_in_ckunk[x + ":" + z].map(({ id }) => id)
      })
```